runtime/src/common/bytes.rs

```diff
@@ -171,37 +175,90 @@ macro_rules! impl_bytes {
                         &self,
                         formatter: &mut ::std::fmt::Formatter,
                     ) -> ::std::fmt::Result {
-                        formatter.write_str("bytes or sequence of u8")
+                        formatter.write_str("bytes or string expected")
                     }
 
-                    fn visit_seq<A>(self, mut seq: A) -> Result<$name, A::Error>
+                    fn visit_str<E>(self, data: &str) -> Result<$name, E>
                     where
-                        A: ::serde::de::SeqAccess<'de>,
+                        E: ::serde::de::Error,
                     {
                         let mut array = [0; $size];
-                        for i in 0..$size {
-                            array[i] = seq
-                                .next_element()?
-                                .ok_or_else(|| ::serde::de::Error::invalid_length(i, &self))?;
+                        let bytes = base64::decode(data).unwrap();
+                        if bytes.len() != $size {
+                            return Err(::serde::de::Error::invalid_length(bytes.len(), &self));
                         }
+                        array[..].copy_from_slice(&bytes);
+
                         Ok($name(array))
                     }
 
-                    fn visit_bytes<E>(self, data: &[u8]) -> Result<$name, E>
+                   fn visit_bytes<E>(self, data: &[u8]) -> Result<$name, E>
                     where
                         E: ::serde::de::Error,
                     {
-                        let mut array = [0; $size];
-                        if data.len() != $size {
+                        if data.len()!=$size {
                             return Err(::serde::de::Error::invalid_length(data.len(), &self));
                         }
+                        let mut array = [0; $size];
                         array[..].copy_from_slice(data);
+
                         Ok($name(array))
                     }
                 }
 
-                Ok(deserializer.deserialize_bytes(BytesVisitor)?)
+                if deserializer.is_human_readable() {
+                    Ok(deserializer.deserialize_string(BytesVisitor)?)
+                } else {
+                    Ok(deserializer.deserialize_bytes(BytesVisitor)?)
+                }
+
             }
         }
     };
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // Use hash of an empty string as a test key.
+    const TEST_KEY_BYTES: [u8; 32] = [
+        0xc6, 0x72, 0xb8, 0xd1, 0xef, 0x56, 0xed, 0x28, 0xab, 0x87, 0xc3, 0x62, 0x2c, 0x51, 0x14,
+        0x06, 0x9b, 0xdd, 0x3a, 0xd7, 0xb8, 0xf9, 0x73, 0x74, 0x98, 0xd0, 0xc0, 0x1e, 0xce, 0xf0,
+        0x96, 0x7a,
+    ];
+
+    #[test]
+    fn test_serde_base64() {
+        // Serialize
```

Comment:
```suggestion
        // Serialize.
```
(also in comments below)